spec/requests/v1/tasks_spec.rb

```diff
@@ -327,4 +327,37 @@
       expect(task).to be_in_progress
     end
   end
+
+  describe 'PATCH /v1/tasks/:task_id/cancel_correction' do
+    it 'deletes incomplete correction submissions and updates the task status' do
+      task = FactoryBot.create(:task, supplier: supplier, status: 'correcting')
+      old_submission_completed = FactoryBot.create(:completed_submission,
+                                                   task: task,
+                                                   created_at: 3.days.ago,
+                                                   aasm_state: 'completed')
+      FactoryBot.create(:submission_with_invalid_entries,
+                        task: task,
+                        created_at: 2.days.ago)
+      FactoryBot.create(:submission_with_validated_entries,
+                        task: task,
+                        created_at: 1.day.ago,
+                        aasm_state: 'in_review')
+
+      patch "/v1/tasks/#{task.id}/cancel_correction", headers: json_headers.merge('X-Auth-Id' => user.auth_id)
+
+      expect(response).to be_successful
+
+      task.reload
+
+      expect(task).to be_completed
+      expect(task.submissions).to eq([old_submission_completed])
+    end
+
+    it 'returns an error if attempting to do this on a non correcting task' do
+      task = FactoryBot.create(:task, supplier: supplier, status: 'completed')
+      expect do
+        patch "/v1/tasks/#{task.id}/cancel_correction", headers: json_headers.merge('X-Auth-Id' => user.auth_id)
+      end.to raise_error(AASM::InvalidTransition)
```

Comment:
This suggests to me that this `else` never gets reached:

https://github.com/dxw/DataSubmissionServiceAPI/pull/328/files#diff-ccc873f4133c19aa7be93616ff932243R65

Is it therefor redundant?

Reply:
Indeed, we can't think of a reason why the else would be reached, and maybe it's more sensible to save! and raise an exception on the API application rather than the inevitable JSON parse error on the Front End (which we don't catch anyway). Happy to do this, would just be a break from how the other methods are implemented.

Reply:
OK, given we are calling a `!` method here then I reckon we should remove the redundant else. I'd prefer clear code without redundant execution paths over consistency.